www/js/lib/uiUtil.js

```diff
@@ -171,6 +171,174 @@ function slideAway (e) {
     }
 }
 
+/*
+  * Returns a list of headings from an article
+    * @param {String} the page for which table of cotents needs to be listed
+    * @returns {List} a list of all headings as objects
+*/
+function HeadingsTOC (articleDoc) {
+    this.doc = articleDoc;
+    this.headings = this.doc.querySelectorAll('h1, h2, h3, h4, h5, h6');
+
+    this.getHeadingObjects = function () {
+        const headings = [];
+        for (let i = 0; i < this.headings.length; i++) {
+            const element = this.headings[i];
+            const obj = {};
+
+            if (element.id) {
+                obj.id = element.id;
+            } else {
+                // generating custom id if id attribute is not present in element
+                const generatedId = element.textContent
+                    .toLowerCase()
+                    .trim()
+                    .replace(/[^\w\s-]/g, '')
+                    .replace(/\s+/g, '-')
+                    .replace(/-+/g, '-');
+                obj.id = 'pph-' + i + '-' + generatedId;
+                element.id = obj.id; // to target the element
+            }
+            obj.index = i;
+            obj.textContent = element.textContent;
+            obj.tagName = element.tagName;
+            headings.push(obj);
+        }
+        return headings;
+    };
+}
+
+/*
+    * Setup table of contents and display the list when the dropup button is clicked
+    * @param {List} a list of all headings as objects
+    * @returns {void}
+*/
+// Inject table of contents list into dropup element and scroll selection into view
```

Comment:
Ditto. Please correct this header. You list a parameter which isn't in the function declaration, for example. Incorporate the comment "// Inject table of contents..." into the JSDoc declaration, rather than having it afterwards. Format correctly!